community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -545,28 +547,29 @@ private void applyRecoveredUpdates() throws IOException, IndexEntryConflictExcep
                     updater.remove( recoveredNodeIds );
```

Comment:
no, not always; for example, we can use LuceneIndexUpdater. Also, this is not related to PR.